packages/TelegramClient-Core.package/TCCUser.class/instance/^equals.st

```diff
@@ -0,0 +1,10 @@
+comparing
+= aUser
+
+	self species = aUser species
+		ifTrue: [
+			^ self firstName = aUser firstName
+			and: [self lastName = aUser lastName]
+			and: [self id = aUser id]
+			and: [self username = aUser username]]
+		ifFalse: [^ false]
```

Comment:
Species is nice :D

Anyway, I would suggest taking the return value out of the blocks and before "self species"

Reply:
we did so because it was done this way in other classes. There seem to be many ways of implementing this test, the next commit will feature a cleaner syntax